### PR TITLE
A few small build and runtime fixes

### PR DESCRIPTION
**Pull request by z33ky**
Just some issues I stumbled upon in a few places.

### CODE REVIEW

File: amnesia/src/game/LuxInputHandler.h [hunk 1]
```diff
@@ -122,6 +122,8 @@ class cLuxInputHandler : public iLuxUpdateable
 
 	void AppDeviceWasPlugged();
 	void AppDeviceWasRemoved();
+#else
+	bool IsGamepadPresent() { return false; }
```
Comment (z33ky) [inline amnesia/src/game/LuxInputHandler.h:126]: The `USE_GAMEPAD` variant could also just be removed altogether...

**Comment by pollend** [comment]
@z33ky can you also make those changes in my bgfx branch: https://github.com/pollend/AmnesiaTheDarkDescent/pull/7

File: HPL2/core/include/system/Platform.h [hunk 2]
```diff
@@ -33,7 +33,11 @@ namespace hpl {
 
 	//-----------------------------------------
 
-    struct VideoComp : public std::binary_function<cVideoMode, cVideoMode, bool> {
+    struct VideoComp
+    #if __cplusplus < 201103L
+        : public std::binary_function<cVideoMode, cVideoMode, bool>
+    #endif
```
Comment (pollend) [inline HPL2/core/include/system/Platform.h:39]: do we need this I think I would just opt to drop this from the code base. I don't think I intend to compile this with c++ 2011.

File: HPL2/tools/editors/common/EditorWindowViewport.cpp [hunk 3]
```diff
@@ -446,7 +446,7 @@ bool cEditorWindowViewport::MenuView_Rendering(iWidget *apWidget, const cGuiMess
 	// Set render mode
 	cWidgetMenuItem* pItem = (cWidgetMenuItem*) apWidget;
 
-	for(int i=0;i<eRenderer_LastEnum;++i)
+	for(int i=0;i<2;++i)
```
[the remark this answers is not in the document]
Comment (pollend) [reply HPL2/tools/editors/common/EditorWindowViewport.cpp:449]: I guess it doesn't hurt anything.

**Comment by pollend** [comment]
very nice, these changes look mostly good.

File: HPL2/core/include/system/Platform.h (hunk 2 above)
Comment (z33ky) [reply HPL2/core/include/system/Platform.h:39]: `std::binary_function` is for *pre* C++11, as in, it has been deprecated with C++11 (note the check does `<`, not `<=`).
I will remove it completely then.

File: HPL2/tools/editors/common/EditorWindowViewport.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (z33ky) [reply HPL2/tools/editors/common/EditorWindowViewport.cpp:449]: Do you want to add the "Simple" render mode instead?
It's more than a single line change, but still not complicated either. I have the diff for that as well, so it'd be no problem to do that instead if you prefer.

**Comment by z33ky** [comment]
> @z33ky can you also make a PR to my bgfx branch: #7
> 
> I intend to switch master to legacy and only use bgfx going forward.

Sure, can do.
When this is merged I would merge master into bgfx, or do you already want to ignore the master branch and just have my changes on the bgfx branch only?

**Comment by pollend** [comment]
I'm going to ignore master

**Comment by z33ky** [comment]
The ParticleViewer and LevelEditor were commented out, but they were easy to fix up and ran.

On the bgfx branch I get a slew of warnings and the binaries don't start properly on my system. The tools seem to still depend on GLEW, which has been disabled, and the Amnesia game binary complains about "Shader model 2 not supported".
I'd open an issue for that, but it seems your repository does not have the "Issues" activated - I'm guessing an oversight due to forking from FrictionalGames, where "Issues" are disable as well? You can add it in https://github.com/pollend/AmnesiaTheDarkDescent/settings.

**Comment by pollend** [comment]
> The ParticleViewer and LevelEditor were commented out, but they were easy to fix up and ran.
> 
> On the bgfx branch I get a slew of warnings when building, mostly deprecation warnings, and the binaries don't start properly on my system. The tools seem to still depend on GLEW, which has been disabled, and the Amnesia game binary complains about "Shader model 2 not supported".
> I'd open an issue for that, but it seems your repository does not have the "Issues" activated - I'm guessing an oversight due to forking from FrictionalGames, where "Issues" are disabled as well? You can add it in https://github.com/pollend/AmnesiaTheDarkDescent/settings.

I've mainly been working on the model viewer and I need to add quite a lot before I can run the game. I'm going to look at the level editor next.

**Comment by z33ky** [comment]
I can confirm the ModelViewer works, minus the texture memory usage display and I suppose some graphical features are missing as of yet, like alpha, specular maps and shadow-casting.
It crashes when exiting because a nullptr is passed at `mpGraphics->DestoroyDepthStencilBuffer(mpDepthStencil[type])` in `cRendererDeferred::DestroyData()`.

**Comment by pollend** [comment]
> I can confirm the ModelViewer works, minus the texture memory usage display and I suppose some graphical features are missing as of yet, like alpha, specular maps and shadow-casting.
> It crashes when exiting because a nullptr is passed at `mpGraphics->DestoroyDepthStencilBuffer(mpDepthStencil[type])` in `cRendererDeferred::DestroyData()`.

They are replaced with rendertarget which is just a device handle and an array of images. I think that is a lot simpler.

**Pull request merged**